### PR TITLE
Latest Departure time

### PR DESCRIPTION
**Pull request by YvonneAriri**
Description:

-Implemented functionality to calculate the latest departure time for the user. The code takes into account rain and high wind speed. if this is detected, an alert is displayed, advising the user to leave 3 hours early and update the latest departure time to 3 hours earlier.
-Implemented the weather feature to show only when the user is hovering over the event.
-Created an "is Loading" feature to enhance user experience. While data is being fetched, a "Loading" message is displayed, indicating that the process is ongoing until the data has been completely fetched.

Milestone Achievement:
Implemented the code to show the latest departure time and updates for an event, considering the weather conditions such as rain or high wind speed three hours before the event starts. Made the weather feature be displayed only when hovering on an event, and created a loading feature to display only while fetching data.

Test Plan:
Eveprep17

### CODE REVIEW

File: event-app-api/JWT.js [hunk 1]
```diff
@@ -1,15 +1,16 @@
 import pkg from "jsonwebtoken";
 
 const { sign, verify } = pkg;
+const secretKey = `${process.env.SECRET_KEY}`;
```
Comment (sophia129) [inline event-app-api/JWT.js:4]: just curious, the process variables are set in the command line when you run the `node` command right?

File: event-app-api/server.js [hunk 2]
```diff
@@ -168,6 +167,16 @@ app.get("/users", async (req, res) => {
   }
 });
 
+//sending a get request to retrieve an event with a specific id
+app.get("/find_event/:id", async (req, res) => {
+  const id = Number(req.params.id);
+  const event = await Event.findOne({
+    order: [["starttime", "ASC"]],
```
Comment (sophia129) [inline event-app-api/server.js:174]: if you're only finding 1 event and the key is unique, you don't need to specify ordering

File: event-app-ui/src/components/Directions/Directions.jsx [hunk 3]
```diff
@@ -1,39 +1,123 @@
...
+          for (let info of data.list) {
+            const hoursFromNowTillEvent = Math.floor(
+              secondsBetweenEpochs(info.dt, result.data.starttime) /
+                SECONDS_IN_AN_HOUR
+            );
+
+            if (
+              hoursFromNowTillEvent === 3 ||
+              hoursFromNowTillEvent === 4 ||
+              hoursFromNowTillEvent === 5
```
Comment (sophia129) [inline event-app-ui/src/components/Directions/Directions.jsx:100]: I think you just want hoursFromNowTillEvent <= 3

**Review by sophia129 — APPROVED** [review]

File: event-app-api/server.js (hunk 2 above)
Comment (YvonneAriri) [reply event-app-api/server.js:174]: okay I'll change that

File: event-app-api/JWT.js (hunk 1 above)
Comment (YvonneAriri) [reply event-app-api/JWT.js:4]: Yes it is

File: event-app-ui/src/components/Directions/Directions.jsx (hunk 3 above)
Comment (YvonneAriri) [reply event-app-ui/src/components/Directions/Directions.jsx:100]: Okay i'll look into that

**Pull request merged**